jobs/etcd/templates/post-start.erb

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash -exu
+
+JOB_DIR=/var/vcap/jobs/etcd/bin
+LOG_DIR=/var/vcap/sys/log/etcd
+CERT_DIR=/var/vcap/jobs/etcd/config/certs
+PKG_DIR=/var/vcap/packages/etcd-consistency-checker
+
+function main() {
+  set +x
+  source "${JOB_DIR}/etcd_bosh_utils.sh"
+  set -x
+
+  local tls_flags
+  tls_flags=""
+
+  <% if p("etcd.require_ssl") %>
+  set +e
+  /var/vcap/packages/etcd-dns-checker/bin/check-a-record <%= p("etcd.dns_health_check_host") %>
+  if [ "0" != "$?" ]; then
+    echo "DNS is not up"
+    exit 1
+  fi
+  set -e
+
+  tls_flags="\
+  --ca-cert ${CERT_DIR}/server-ca.crt \ 
+  --cert ${CERT_DIR}/client.crt \
+  --key ${CERT_DIR}/client.key"
+  <% end %>
+
+
+  set +e
+  local consistency_check_output
+  consistency_check_output=$(timeout 10 "${PKG_DIR}/bin/etcd-consistency-checker" \
+    --cluster-members "${consistency_checker_cluster_members}" \
+    ${tls_flags})
+  exit_code="$?"
+  set -e
+
+  if [ "124" != "$exit_code" ] && [ "0" != "$exit_code" ]; then
```

Comment:
Why is timeout ok here?

Reply:
The consistency checker runs on a constant loop if it succeeds